src/core/server/rendering/rendering_service.tsx

```diff
@@ -164,12 +148,163 @@ export class RenderingService {
 
   public async stop() {}
 
+  /**
+   * Assign values for branding related configurations based on branding validation
+   * by calling checkBrandingValid(). For dark mode URLs, add additonal validation
+   * to see if there is a valid default mode URL exist first. If URL is valid, pass in
+   * the actual URL; if not, pass in undefined.
+   *
+   * @param {boolean} darkMode
+   * @param {Readonly<OpenSearchDashboardsConfigType>} opensearchDashboardsConfig
+   * @returns {BrandingAssignment} valid URLs or undefined assigned for each branding configs
+   */
+  private assignBrandingConfig = async (
+    darkMode: boolean,
+    opensearchDashboardsConfig: Readonly<OpenSearchDashboardsConfigType>
+  ): Promise<BrandingAssignment> => {
+    const brandingValidation: BrandingValidation = await this.checkBrandingValid(
+      darkMode,
+      opensearchDashboardsConfig
+    );
+    const branding = opensearchDashboardsConfig.branding;
+
+    // assign default mode URL based on the brandingValidation function result
+    const logoDefault = brandingValidation.isLogoDefaultValid
+      ? branding.logo.defaultUrl
+      : undefined;
+
+    const markDefault = brandingValidation.isMarkDefaultValid
+      ? branding.mark.defaultUrl
+      : undefined;
+
+    const loadingLogoDefault = brandingValidation.isLoadingLogoDefaultValid
+      ? branding.loadingLogo.defaultUrl
+      : undefined;
+
+    // assign dark mode URLs based on brandingValidation function result
+    let logoDarkmode = brandingValidation.isLogoDarkmodeValid
+      ? branding.logo.darkModeUrl
+      : undefined;
+
+    let markDarkmode = brandingValidation.isMarkDarkmodeValid
+      ? branding.mark.darkModeUrl
+      : undefined;
+
+    let loadingLogoDarkmode = brandingValidation.isLoadingLogoDarkmodeValid
+      ? branding.loadingLogo.darkModeUrl
+      : undefined;
+
+    /**
+     * For dark mode URLs, we added another validation:
+     * user can only provide a dark mode URL after providing a valid default mode URL,
+     * If user provides a valid dark mode URL but fails to provide a valid default mode URL,
+     * return undefined for the dark mode URL
+     */
+    if (logoDarkmode && !logoDefault) {
+      this.logger
+        .get('branding')
+        .error('Must provide a valid logo default mode URL before providing a logo dark mode URL');
```

Comment:
We should gather a list of our errors and perhaps we can check with the writing team to verify the error messages is ok.

Reply:
Good idea. I can organize a complete list.